execution_engine_testing/test_support/src/wasm_test_builder.rs

```diff
@@ -576,6 +577,37 @@ where
         let engine_state = Rc::get_mut(&mut self.engine_state).unwrap();
         engine_state.update_config(engine_config);
 
+        let empty_path: Vec<String> = vec![];
+
+        if let Ok(StoredValue::CLValue(cl_registry)) = self.query(
+            self.post_state_hash,
+            Key::SystemContractRegistry,
+            &empty_path,
+        ) {
+            let registry = CLValue::into_t::<SystemContractRegistry>(cl_registry).unwrap();
+            if self.mint_contract_hash.is_none() {
+                self.mint_contract_hash = Some(*registry.get(MINT).expect("should have mint hash"))
+            };
+            if self.handle_payment_contract_hash.is_none() {
+                self.handle_payment_contract_hash = Some(
+                    *registry
+                        .get(HANDLE_PAYMENT)
+                        .expect("should have handle payment hash"),
+                )
+            }
+            if self.standard_payment_hash.is_none() {
+                self.standard_payment_hash = Some(
+                    *registry
+                        .get(STANDARD_PAYMENT)
+                        .expect("should have standard payment hash"),
+                )
+            }
+            if self.auction_contract_hash.is_none() {
+                self.auction_contract_hash =
+                    Some(*registry.get(AUCTION).expect("should have auction hash"))
+            }
+        }
```

Comment:
It's not clear to me why it worked previously and doesn't do anymore. What has changed that we now have to populate `self.*` manually?

Reply:
This is to compensate for introducing the system contract registry in an upgrade request, which we weren't running into our earlier test fixtures